content_handler/application_impl.cc

```diff
@@ -6,25 +6,46 @@
 
 #include <utility>
 
+#include "apps/dart_content_handler/zip/unzipper.h"
+#include "flutter/content_handler/runtime_holder.h"
 #include "lib/ftl/logging.h"
 #include "mojo/public/cpp/application/connect.h"
 
 namespace flutter_content_handler {
+namespace {
+
+constexpr char kSnapshotKey[] = "snapshot_blob.bin";
+
+std::vector<char> ExtractSnapshot(std::vector<char> bundle) {
+  zip::Unzipper unzipper(std::move(bundle));
+  return unzipper.Extract(kSnapshotKey);
+}
+
+}  // namespace
 
 ApplicationImpl::ApplicationImpl(
     mojo::InterfaceRequest<mojo::Application> application,
     mojo::URLResponsePtr response)
-    : binding_(this, std::move(application)),
-      initial_response_(std::move(response)) {}
+    : binding_(this, std::move(application)) {
+  drainer_.reset(new glue::DrainDataPipeJob(
```

Comment:
Why keep the drainer around as an ivar when you can get  rid of it in the drain jobs callback itself? I seem to recall us doing something similar elsewhere.